src/modules/user/user.service.ts

```diff
@@ -268,33 +269,72 @@ export class UserService {
   }
 
   async getFheelancerInformation(
-    take: number,
-    page: number,
-    search: string,
+    params: GetFreelancerDto,
   ): Promise<[User[], number]> {
     try {
-      const currentUser = await this.userRepository
+      const {
+        page,
+        take,
+        skills,
+        categories,
+        hourly_rate_start,
+        hourly_rate_end,
+        search,
+        ...userPayload
+      } = params;
+      const role = 'Freelancer';
+
+      const query = this.userRepository
         .createQueryBuilder('user')
         .leftJoinAndSelect('user.category', 'category')
+        .where(userPayload)
+        .andWhere('user.role >= :role', {
+          role,
+        })
         .skip((page - 1) * take)
-        .take(take)
-        .where(
+        .take(take);
+
+      if (hourly_rate_start) {
+        query.andWhere('user.hourly_rate >= :hourly_rate_start', {
+          hourly_rate_start,
+        });
+      }
+
+      if (hourly_rate_end) {
+        query.andWhere('user.hourly_rate <= :hourly_rate_end', {
+          hourly_rate_end,
+        });
+      }
+
+      if (search) {
+        query.andWhere(
           new Brackets((qb) => {
             qb.where('user.category LIKE :search')
               .orWhere('user.position LIKE :search')
               .orWhere('user.hourly_rate LIKE :search')
               .orWhere('user.available_time LIKE :search', {
```

Comment:
Like is a very very expensive operation, it should be used only for search, in your case its filter

Reply:
fixed